alchemist-composeui/src/commonMain/kotlin/it/unibo/alchemist/boundary/composeui/App.kt

```diff
@@ -9,37 +9,127 @@
-        Column(Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
-            Button(onClick = { showContent = !showContent }) {
-                Text("Click me!")
-            }
-            AnimatedVisibility(showContent) {
-                val greeting = remember { getPlatform() }
-                Column(Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
-                    Text("Compose: $greeting")
+fun app(viewModel: SimulationViewModel = viewModel { SimulationViewModel() }) {
+    val status by viewModel.status.collectAsStateWithLifecycle()
+    val errors by viewModel.errors.collectAsStateWithLifecycle()
+    val nodes by viewModel.nodes.collectAsStateWithLifecycle()
+    Scaffold(
+        topBar = { topBar(status) },
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier.padding(innerPadding).padding(horizontal = 8.dp, vertical = 16.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+        ) {
+            controlButton(status, viewModel::play, viewModel::pause)
+            OutlinedCard(
+                modifier = Modifier.fillMaxSize(),
+                colors = CardDefaults.cardColors(
+                    containerColor = MaterialTheme.colorScheme.surface,
+                ),
+                border = BorderStroke(1.dp, Color.Black),
+            ) {
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(4.dp),
+                    modifier = Modifier.verticalScroll(
+                        rememberScrollState(),
+                    ),
+                ) {
+                    errorDialog(viewModel::fetch, errors)
+                    for (node in nodes) {
+                        nodeDrawer(node.id)
+                    }
                 }
             }
         }
     }
 }
+
+/**
+ * Top bar put on top of the application.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun topBar(status: SimulationStatus) {
+    TopAppBar(
+        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.primaryContainer,
+            titleContentColor = MaterialTheme.colorScheme.primary,
+        ),
+        title = {
+            Text(
+                "Simulation: $status",
+            )
+        },
+    )
+}
+
+/**
+ * Button to control the simulation.
+ */
+@Composable
+fun controlButton(status: SimulationStatus, resume: () -> Unit, pause: () -> Unit) {
+    if (status == SimulationStatus.Running) {
+        Button(onClick = { pause() }) {
+            Text("Pause", modifier = Modifier.padding(8.dp))
+        }
+    } else {
+        Button(onClick = { resume() }) {
+            Text("Resume", modifier = Modifier.padding(8.dp))
+        }
+    }
+}
+
+/**
+ * Display the error dialog, currently used to circumvent the null issue we're facing when subscribing to simulation.
+ */
+@Composable
```

Comment:
It is recommended that the composeable function be capitalized: ErrorDialog

Reply:
I suggest taking a look at https://detekt.dev/docs/introduction/compose/